simulation_server.py

```diff
@@ -153,41 +153,48 @@ def setup_project(self):
             logging.error(f'Unsupported URL protocol: {self.url}')
             return False
         if 'allowedRepositories' not in config or not self.url.startswith(tuple(config['allowedRepositories'])):
+            error = ''
             if not config['docker']:
-                logging.error('Docker not enabled: cannot host foreign simulations.')
-                return False
-            if config['shareIdleTime'] == 0:
-                logging.error('This simulation server is not configured to share idle time.')
-                return False
-            if current_load > config['shareIdleTime']:
-                logging.error(f'Cannot share idle time when current load is above threshold: {current_load}.')
-                return False
-            if 'blockedRepositories' in config and self.url.startswith(tuple(config['bannedRepositories'])):
-                logging.error(f'Cannot run simulation from blocked repository: {self.url}')
-                return False
-
+                error = 'Docker not enabled: cannot host simulations from repositories outside of "allowedRepositories".'
+            elif config['shareIdleTime'] == 0:
+                error = 'This simulation server is not configured to share idle time.'
+            elif current_load > config['shareIdleTime']:
+                error = f'Cannot share idle time when current load is above threshold: {current_load}.'
+            elif 'blockedRepositories' in config and self.url.startswith(tuple(config['blockedRepositories'])):
+                error = f'Cannot run simulation from blocked repository: {self.url}'
```

Comment:
Single initialization:
```suggestion
                error = f'Cannot run simulation from blocked repository: {self.url}'
            else:
                error = False
```